lib/stud/task.rb

```diff
@@ -1,8 +1,21 @@
 require "thread"
 
 module Stud
+
+  # A Task spawns a thread to execute the given block. execution completion and result retrieval is
+  # done using the Task#wait method. A Task in run once and the thread exists upon block completion.
+  # A task and its underlying thread are not reusable.
+  #
+  # Task does not provide a mean to force-interrupt a running task, all it will do is to take note
+  # that an interruption is requested and store that information in the task threadlocal storage
+  # in the {Stud::STUD_STOP_REQUESTED => true} namespace. Any task can verify this from within its
```

Comment:
I'd prefer if we only documented The One True interface for asking if we shoudl stop, so point users at `Stud.stop?` instead of describing the implementation (Thread local storage)